docs/40_smart-contract-guides/80_linking-custom-permission.md

```diff
@@ -2,72 +2,585 @@
-## Step 2. Link Authorization to Your Custom Permission
-Link the authorization to invoke the `upsert` action with the newly created permission:
+```console
+executed transaction: 43b6ad4ce7a52d7281ccd2800caa02d5278ee714de36fefe9624bff621378402  104 bytes  129 us
+#      scholder <= scholder::what               {"user":"name"}
+>> why not name
+```
 
 ```shell
-cleos set action permission alice addressbook upsert upsert
+cleos push action scholder how '["name"]' -p bob@customp1
+```
+The output should be similar to:
+
+```console
+executed transaction: 43b6ad4ce7a52d7281ccd2800caa02d5278ee714de36fefe9624bff621378402  104 bytes  129 us
+#      scholder <= scholder::how               {"user":"name"}
+>> how are you name
+```
+
+We see that the _customp1_ permission can sucessfully call the _what_ and _how_ actions, but fails to call the _why_ action.
+
+Call the actions using the _customp2_ permission:
+
+```shell
+cleos push action scholder why '["name"]' -p bob@customp2
+```
+
+The output should be similar to:
+
+```console
+Error 3090005: Irrelevant authority included
+Please remove the unnecessary authority from your action!
+Error Details:
+action declares irrelevant authority '{"actor":"bob","permission":"customp2"}'; minimum authority is {"actor":"bob","permission":"active"}
 ```
-In this example, we link the authorization to the `upsert` action created earlier in the addressbook contract.
-## Step 3. Test it
-Let's try to invoke the action with an `active` permission:
 
 ```shell
-cleos push action addressbook upsert '["alice", "alice", "liddel", 21, "Herengracht", "land", "dam"]' -p alice@active
+cleos push action scholder what '["name"]' -p bob@customp2
 ```
-You should see an error like the one below:
 
-```text
+The output should be similar to:
+
+```console
 Error 3090005: Irrelevant authority included
 Please remove the unnecessary authority from your action!
 Error Details:
-action declares irrelevant authority '{"actor":"alice","permission":"active"}'; minimum authority is {"actor":"alice","permission":"upsert"}
+action declares irrelevant authority '{"actor":"bob","permission":"customp2"}'; minimum authority is {"actor":"bob","permission":"customp1"}
+```
+
+```shell
+cleos push action scholder how '["name"]' -p bob@customp2
+```
+
+The output should be similar to:
+
+```shell
+executed transaction: 43b6ad4ce7a52d7281ccd2800caa02d5278ee714de36fefe9624bff621378402  104 bytes  129 us
+#      scholder <= scholder::how               {"user":"name"}
+>> how are you name
 ```
-Now, try the **upsert** permission, this time, explicitly declaring the **upsert** permission we just created: (e.g. `-p alice@upsert`)
 
-```text
-cleos push action addressbook upsert '["alice", "alice", "liddel", 21, "Herengracht", "land", "dam"]' -p alice@upsert
+We see that the _customp2_ permission can sucessfully call the _how_ action, but fails to call the _why_ and _what_ actions.
```

Comment:
For consistency with previous instances:
\_how\_ -> \`how\`
\_why\_ and \_what\_ -> \`why\` and \`what\`